python-lib/spell_checker/input_output_params.py

```diff
@@ -0,0 +1,110 @@
+import dataiku
+from dataiku.customrecipe import *
+import re
+
+
+def get_input_output():
+    input_dataset = get_input_names_for_role("input_dataset")[0]
+    not_to_be_corrected_dataset = get_input_names_for_role("not_to_be_corrected_dataset")
+
+    not_to_be_corrected_dataset = NotToBeCorrectedDataset(not_to_be_corrected_dataset)
+
+    if not_to_be_corrected_dataset.is_present:
+        not_to_be_corrected_dataset.check()
+
+    output_dataset = get_output_names_for_role("output_dataset")[0]
+
+    return input_dataset, not_to_be_corrected_dataset, output_dataset
+
+
+def get_spell_checker_params(recipe_config, not_to_be_corrected_dataset):
+    def _p(param_name, default=None):
+        return recipe_config.get(param_name, default)
+
+    params = {}
+
+    # General parameters
+    params["text_col_list"] = _p("text_col_list")
```

Comment:
Streamlined version: hold your list of parameter names in a CONSTANT, and iterate over the list to assign it to a param dictionary